frontends/common/constantFolding.h

```diff
@@ -28,11 +28,11 @@ namespace P4 {
 class DoConstantFolding : public Transform {
  protected:
     const ReferenceMap* refMap;  // if null no 'const' values can be resolved
-    TypeMap* typeMap;  // if null we have no types; updated for new constants
+    const TypeMap* typeMap;  // if null we have no types; updated for new constants
     bool typesKnown;
     bool warnings;  // if true emit warnings
     // maps expressions and declarations to their constant values
-    std::map<const IR::Node*, const IR::Expression*> constants;
+    std::map<const IR::Declaration_Constant*, const IR::Expression*> constants;
```

Comment:
Also, please update the reference outputs. This can be done by running each failed test with the -f flag - after you made sure that the produced output is the intended one. For example:
`./p4/testdata/p4_16_samples/stack.p4.test -f` will update the output produced by this test.